blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxMetadataIO.java

```diff
@@ -0,0 +1,162 @@
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.codahale.metrics.Timer;
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.querybuilder.Insert;
+import com.datastax.driver.core.querybuilder.Select;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import com.rackspacecloud.blueflood.io.CassandraModel;
+import com.rackspacecloud.blueflood.io.Instrumentation;
+import com.rackspacecloud.blueflood.io.MetadataIO;
+import com.rackspacecloud.blueflood.io.serializers.metrics.StringMetadataSerDes;
+import com.rackspacecloud.blueflood.types.Locator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.*;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.*;
+
+/**
+ * Read/write to metrics_metadata using Datastax driver.
+ */
+public class DatastaxMetadataIO implements MetadataIO {
+
+    public static final String KEY = "key";
+    public static final String COLUMN1 = "column1";
+    public static final String VALUE = "value";
+
+    private static final Logger LOG = LoggerFactory.getLogger( DatastaxMetadataIO.class );
+
+    private final StringMetadataSerDes serDes = new StringMetadataSerDes();
+
+    @Override
+    public Map<String, String> getAllValues( Locator locator ) throws IOException {
+
+        Timer.Context ctx = Instrumentation.getReadTimerContext( CassandraModel.CF_METRICS_METADATA_NAME );
+
+        Session session = DatastaxIO.getSession();
+
+        try {
+
+            Select.Where select = select()
+                    .all()
+                    .from( CassandraModel.CF_METRICS_METADATA_NAME )
+                    .where( eq( KEY, locator.toString() ));
+
+            List<Row> results = session.execute( select ).all();
+
+            Map<String, String> values = new HashMap<String, String>();
+
+            for ( Row row : results ) {
+                if ( LOG.isDebugEnabled() ) {
```

Comment:
I dont see much value in those log statements but I would be okay with 'trace' level logging.